### ISSUE TITLE
Top Posts & Pages: inconsistent behaviour with block availability

### ISSUE THREAD

**Issue by monsieur-z** [Automattic/jetpack#38695]
### Impacted plugin

Jetpack

### Quick summary

I created two new self-hosted Jetpack sites with a Free plan, both using Jurassic Ninja. The Jetpack version of one of them is controlled by the Beta Tester plugin. Though both sites run on Jetpack 13.7-a.7, the block is not available on the site controlled by the Beta Tester.

This issue is strange, both sites look identical in every way. I can't pinpoint why the block is available in one case and not the other.

### Steps to reproduce

- Create a new site with Jurassic Ninja, not using the Beta Tester
- Connect it
- Create a new post
- Check that the Top Posts & Pages block is available
- Repeat the operation, this time controlling the Jetpack version with the Beta Tester plugin
- Notice the block is not available

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by jeherve** [comment]
> this time controlling the Jetpack version with the Beta Tester plugin

Could you clarify how you went about doing that in practice? Did you start a brand new JN site, but checking to enable the Beta tester plugin? Did you select a specific branch? Did you connect the new site to your WordPress.com account after creating it? Was the Stats module active on the site?

**Comment by jp-imagines** [comment]
I came across a potentially related issue when I was looking into https://github.com/Automattic/jetpack/issues/32937#issuecomment-2270133971, where the Top Posts & Pages block wasn't available on a Simple site (the classic widget wasn't available either, which is expected).

On a quick check across a few test sites on Dotcom, I'm only seeing the Top Posts & Pages block available on a WoA site, but not Simple sites. (For reference, I can add it on `wootestjpi.blog` (WoA, Coming Soon, Stats active), but not `premium.wootestjpi.blog` (Simple, Public).)

**Comment by Aurorum** [comment]
The block was removed for Simple sites in https://github.com/Automattic/jetpack/pull/38564, so I'm afraid that's intentional (the API was broken on Simple sites, and meant that the block never properly worked for them). I've proposed bringing back the widget in https://github.com/Automattic/jetpack/pull/38610 for the meantime.

**Comment by monsieur-z** [comment]
Looking further into this, I noticed it happened with a couple of other blocks (Jetpack AI Search, Subscribe, and Blog Stats). That led me to check the connection to Jetpack, and indeed, in one instance, my user account was not connected.

I was thrown off by the different connection flows in one case and the other. I'm not sure why that was since both test sites ran 13.7-beta.

Closing since this is a non-issue.
